### PR TITLE
zplug: Add initial support for zplug package manager

### PR DESCRIPTION
**Pull request by s1341**
Hi,

I've added support for the zplug package manager for zsh.

I'd be grateful for any feedback.
s1341

### CODE REVIEW

File: modules/programs/zplug.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        type = types.listOf types.str;
+        default = [ ];
+        description = "The plugin tags";
+      };
+    };
+
+  });
+
+in {
+  options.programs.zplug = {
```
Comment (rycee) [inline modules/programs/zplug.nix]: I wonder if it wouldn't make sense to put this under `programs.zsh.zplug`?

File: modules/programs/zplug.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    enable = mkEnableOption "zplug - a zsh plugin manager";
+
+    plugins = mkOption {
+      default = [ ];
+      type = types.listOf pluginModule;
+      description = ''
+        List of zplug plugins
```
Comment (rycee) [inline modules/programs/zplug.nix]:
```suggestion
        List of zplug plugins.
```

File: modules/programs/zplug.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        type = types.str;
+        description = ''
+          The name of the plugin.
+        '';
+      };
+
+      tags = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = "The plugin tags";
```
Comment (rycee) [inline modules/programs/zplug.nix]:
```suggestion
        description = "The plugin tags.";
```

File: modules/programs/zplug.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+with lib;
+
+let
+
+  cfg = config.programs.zplug;
+
+  pkg = pkgs.srcOnly {
+    name = "zplug";
+    src = builtins.fetchGit {
```
Comment (rycee) [inline modules/programs/zplug.nix]: This is not reproducible and it also introduces a package to Home Manager. I much rather you'd add `zplug` as a package in https://github.com/NixOS/nixpkgs/ as a proper package with version, license information, and such. Once it is added to Nixpkgs you can refer to it through `pkgs.zplug`.

**Comment by rycee** [comment]
Thanks for the contribution! I've added a few comments.

**Comment by s1341** [comment]
> Thanks for the contribution! I've added a few comments.

I'll get on those. I'll start with a PR for zplug in nixpkgs. That might take a few days to get through their process.

**Comment by s1341** [comment]
zplug PR in nixpkgs: https://github.com/NixOS/nixpkgs/pull/83856

**Comment by s1341** [comment]
@rycee I've resolved all the issues you mentioned. Do you want to take another look?

**Comment by rycee** [comment]
Thanks, I added a test case and rebased to master in 1b210e7143547ce0f41e8082b8d27e9c7d220351.

**Pull request closed without merging**